src/history/hs_rec.c

```diff
@@ -438,22 +425,19 @@ __wt_hs_insert_updates(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_MULTI *mult
          * 4) We have a single tombstone on the chain, it is simply ignored.
          */
         squashed = false;
-        for (upd = list->onpage_upd, non_aborted_upd = prev_upd = NULL; upd != NULL;
-             prev_upd = non_aborted_upd, upd = upd->next) {
+        for (upd = list->onpage_upd, prev_upd = NULL; upd != NULL; upd = upd->next) {
             if (upd->txnid == WT_TXN_ABORTED)
                 continue;
 
-            non_aborted_upd = upd;
-
-            /* Detect out of order timestamp update. */
-            if (min_ts_upd != NULL && min_ts_upd->start_ts < upd->start_ts &&
-              out_of_order_ts_upd != min_ts_upd) {
+            /* Detect any out of order timestamp update. */
+            if (prev_upd != NULL && prev_upd->start_ts < upd->start_ts) {
+                WT_ASSERT(session, prev_upd->start_ts == WT_TS_NONE);
                 /*
                  * Fail the eviction if we detect out of order timestamps and the error flag is set.
```

Comment:
Here as well

Reply:
May be do a global search of out of order

Reply:
Searched all the out of order strings and corrected those that are applicable.